torch/_functorch/_aot_autograd/subclass_utils.py

```diff
@@ -70,33 +109,23 @@ def create_subclass_metadata(a: Any, start_idx: int, count_symints: bool):
             outer_size=a.size(),  # type: ignore[attr-defined, arg-type]
             outer_stride=a.stride(),  # type: ignore[arg-type]
             original_subclass=a,
+            memory_format=maybe_suggest_memory_format(a, with_memory_format),
         ),
         new_start_idx,
     )
 
 
-# Given a real tensor subclass, returns a nested list of Plain tensor types
-def get_types_for_subclass(tensor_subclass):
-    if not is_traceable_wrapper_subclass(tensor_subclass):
-        return ["Tensor"]
-    inner_keys, _ = tensor_subclass.__tensor_flatten__()
-    result = []
-    for key in inner_keys:
-        inner_tensor = getattr(tensor_subclass, key)
-        result.extend(get_types_for_subclass(inner_tensor))
-    return result
-
-
 # Given a flat list of arguments, some of which may be tensor subclasses,
 # computes metadata about "how to reconstruct the current list of subclasses,
 # if we were given their flattened dense tensors instead"
 def create_subclass_meta(
     curr_args: Union[List[Any], Tuple[Any, ...]],
     *,
     count_symints: bool = True,
-) -> List[Union[int, SubclassCreationMeta]]:
+    with_memory_format: bool = False,
```

Comment:
Can you help me understand why we want to sometimes **not** include memory_format  when creating subclass meta? If there is a good reason for doing it sometimes and not others, a comment explaining exactly when it is / is not necessary would be nice.

Reply:
My main logic was to not add overhead on deducing memory_format.
This  could also be especially painful if to call it during tracing on FakeTensors with symbolic shapes - memory format checks in my experience  give hairy symbolic shapes checks on strides (divisibility, equal to 1, reminder equals 0 etc.).

We use create_subclass_meta for inputs, outputs (in collect_metadata_analysis). I have not seen any usage of memory_format for inputs?

If we need memory format for inputs and outputs too - we can make it non-optional.

Reply:
oh that's fair - we don't need the memory format info for inputs. Can you just mention that in a comment?